File: lib/controllers/leave.js
const _ = require('lodash');
const leave = require('../pairrit').leave;

module.exports = (req, res) => {
  const userName  = req.body.user_name;
  const channelId = req.body.channel_id;

  leave(userName, channelId)
    .then((pairName) => {
      if (_.isUndefined(pairName)) {
        res.send('You are not currently in a pair. Use `/pair join` to join one!');
      } else {
        res.send(`You have left the \`${pairName}\` pair.`);
      }
    });
};

File: app/models/command/leave.rb
class Command::Leave
  attr_reader :channel, :params, :user

  def initialize(params)
    @params = params
    @user = params['user']
    @channel = params['channel']
  end

  def process
    pair = user.leave(channel)

    if pair
      { text: "You have left the `#{pair.name}` pair." }
    else
      { text: 'You are not in a pair. Why not try `/pair join` to join one?' }
    end
  end
end
